### ISSUE TITLE
Websites where the Add Custom Search fails

### ISSUE THREAD

**Issue by ssborbis** [ssborbis/ContextSearch-web-ext#67]
If you come across any sites / forms that don't work with the Add Custom Search function, list them here. I have some workarounds in mind, but I need more examples. 

news.google.com - main search form

**Comment by ssborbis** [comment]
> Note: The "Find" button in advanced mode (search on MycroftProject) doesn't work (version 1.6.3).

Thanks. 1.6.3.1 going up. Dunno how some of this stuff gets by me and the linter

**Comment by ssborbis** [comment]
Ok, new code works for those websites. Should work with all non-standard search forms using GET

**Comment by DenB10** [comment]
Works fine :) 
But  tooltip "PerformASearchTooltip" doesn't appear!

**Comment by ssborbis** [comment]
> But tooltip "PerformASearchTooltip" doesn't appear!
This is driving me nuts. I can't find the problem.

**Comment by ssborbis** [comment]
nevermind, got it

**Comment by sojusnik** [comment]
http://context.reverso.net/translation/

**Comment by ssborbis** [comment]
> http://context.reverso.net/translation/

Fix ready

**Comment by sojusnik** [comment]
https://www.similarweb.com/

**Comment by ssborbis** [comment]
Same fix seems to work for that one too. I'll get a new release in the next couple days

**Comment by Tubelerczyk** [comment]
After updating Firefox has stopped working. Previously it worked:

[http://chomikuj.pl/action/SearchFiles](http://chomikuj.pl/action/SearchFiles)


![screen](https://user-images.githubusercontent.com/41324346/44196586-9eda0c80-a13c-11e8-8572-a14ed348d719.jpg)

--------------------------------------------------------------------------------------

This, for example, does not work either:

[https://www.gry-online.pl/szukaj.asp](https://www.gry-online.pl/szukaj.asp)

![screen2](https://user-images.githubusercontent.com/41324346/44196841-54a55b00-a13d-11e8-882a-3850dc27b5da.jpg)

**And there are more such search engines.**

**Comment by sojusnik** [comment]
Can confirm that with ContextSearch 1.7.0 on Firefox 61.0.1.

While all OpenSearch engine work fine, f.i. when searching from the address bar, some of the ContextSearch ones aren't. [Here's](https://i.imgur.com/4QrlG9h.png) a screenshot of one of them.

**Comment by Tubelerczyk** [comment]
I have Firefox version 61.0.2 (64bit). The 1.6.9 version of the plugin works well on this Firefox. Version 1.7.0 does not support some search engines. For now I am using version 1.6.9.

**Comment by ssborbis** [comment]
hotfix going up 1.7.0.1

**Comment by GlasWolf** [comment]
https://who-called.co.uk/

**Comment by ssborbis** [comment]
> https://who-called.co.uk/

That site adds ok for me. It uses the alternative 'do a search' method, but it added. What happens when you try?

**Comment by GlasWolf** [comment]
That's odd - it worked this time.  When I tried earlier it threw up an error, very roughly "ContextSearch was unable to create a search entry for that website".

Not wanting to hijack this thread but is there any way to simulate a search done from the search box, rather than simply insert the search term into the URL?  Phone numbers are usually shown formatted with spaces, so that method doesn't work.

**Comment by GlasWolf** [comment]
Basically I want to be able to highlight and ContextSearch phone numbers which include hyphens and spaces.  The {searchTerms} method gives a 404 but, as you say, the site's own search function automatically strips them.  Can that be replicated somehow, maybe using POST?

**Comment by ssborbis** [comment]
Ah, I see. I just tried a highlight search for `1-234-567-8910` and got the 404 error. I'd have to write some site-specific code or add an option for each engine to modify the search terms with a regex rule. Doable, but a bit time-consuming for a rare issue. 

You might want to enable 'Manage With Bookmarks' and add a bookmarklet to the ContextSearch Menu folder with the following code as the URL. It won't be 100% what you want, but you'll be able to access it via ContextSearch

`javascript:q = "" + (window.getSelection ? window.getSelection() : document.getSelection ? document.getSelection() : document.selection.createRange().text); if (q!=null) window.open("https://who-called.co.uk/Number/" + escape(q).replace(/[^0-9]/g, ""), "_blank"); void 0`

or 

`javascript:q = "" + (window.getSelection ? window.getSelection() : document.getSelection ? document.getSelection() : document.selection.createRange().text); if (q!=null) window.open("https://who-called.co.uk/Number/" + escape(q).replace(/[^0-9]/g, ""), "_blank"); void 0`

for opening in a new tab (popup blocker risk)

**Comment by GlasWolf** [comment]
Great idea to use bookmarklets.  Edited the replace term to `replace(/%25\d\d|[^\d]/g, "")` to remove character codes, and it works perfectly - thanks so much for your help!

**Comment by ssborbis** [comment]
@GlasWolf I've added a regex option to modify search terms for each engine. Look for it in the next release.

**Comment by GlasWolf** [comment]
Even easier!  I'll check it out when it appears, thanks again.

**Comment by ssborbis** [comment]
New regex option is live in 1.8.1. I tested the search engine for https://who-called.co.uk/ using the search regex `"%25\\d\\d|[^\\d]", ""` and it seems to work. I didn't actually get a proper result due to my ip being blocked (VPN) but the url looked good

searched for `1-234-567-8910`
resulting url `https://who-called.co.uk/Number/12345678910`

**Comment by ebluser** [comment]
This professional medicines search engine uses the POST method. Would it be possible to add it to your complement? Thank you
[https://cima.aemps.es/cima/publico/buscadoravanzado.html](https://cima.aemps.es/cima/publico/buscadoravanzado.html)

**Comment by ssborbis** [comment]
@ebluser 

I looked at https://cima.aemps.es/cima/publico/buscadoravanzado.html and I don't think it can work with ContextSearch. It seems to use javascript to fetch .json using a REST API. 

You could, for example, use a template like this:

`https://cima.aemps.es/cima/rest/medicamentos?multiple2={searchTerms}&cargaprincipiosactivos=true&nomostrarip=1&comerc=1`

... and search for `ABFENTIQ` to receive this result: https://cima.aemps.es/cima/rest/medicamentos?multiple2=ABFENTIQ&cargaprincipiosactivos=true&nomostrarip=1&comerc=1

... but that .json may not be helpful to you.

**Comment by ebluser** [comment]
> 
> 
> @ebluser
> 
> I looked at https://cima.aemps.es/cima/publico/buscadoravanzado.html and I don't think it can work with ContextSearch. It seems to use javascript to fetch .json using a REST API.
> 
> You could, for example, use a template like this:
> 
> `https://cima.aemps.es/cima/rest/medicamentos?multiple2={searchTerms}&cargaprincipiosactivos=true&nomostrarip=1&comerc=1`
> 
> ... and search for `ABFENTIQ` to receive this result: https://cima.aemps.es/cima/rest/medicamentos?multiple2=ABFENTIQ&cargaprincipiosactivos=true&nomostrarip=1&comerc=1
> 
> ... but that .json may not be helpful to you.
 Indeed, it is not. However, ContestSearch has been able to include other search engines with post method. If you could define technical specifications or prerequisites for ContestSearch, it might be useful if the search engine developers could include improvements.
Thanks!

**Comment by ssborbis** [comment]
@ebluser 

> Indeed, it is not. However, ContestSearch has been able to include other search engines with post method. If you could define technical specifications or prerequisites for ContestSearch, it might be useful if the search engine developers could include improvements.
Thanks!

I'm not seeing where that engine uses POST. I'm looking at the network logs and only see GET requests to the REST API. Do you have a POST request I could look at?

**Comment by githubeing** [comment]
> If you come across any sites / forms that don't work with the Add Custom Search function, list them here.

yandex.translate:
1. [entered text](https://translate.yandex.com/?lang=en-fr&text=If%20you%20come%20across%20any%20sites%20%2F%20forms%20that%20don%27t%20work)
2. [web page](https://translate.yandex.com/translate?url=https%3A%2F%2Fgithub.com%2Fssborbis%2FContextSearch-web-ext%2Fissues%2F67&lang=en-fr)

**Comment by githubeing** [comment]
the ocr feature also doesn't work with contextsearch, but not sure if it is possible at all to make it work together. because yandex require to upload the image from local computer, doesn't allow to paste url of image: https://translate.yandex.com/ocr

**Comment by ssborbis** [comment]
@githubeing 

Looks like yandex is using some javascript-based forms I can't account for. Making a custom engine would be trivial though, based on the results url. 

Template: 
`https://translate.yandex.com/?lang=en-fr&text={seachTerms}`

The OCR feature looks difficult to handle. There might be a bookmarlet-based solution though.

**Comment by githubeing** [comment]
> If you come across any sites / forms that don't work with the Add Custom Search function, list them here.

another example: https://yandex.ru/images/ - search by image url (see screenshot)![Screenshot from 2019-03-14 02-08-58](https://user-images.githubusercontent.com/29702626/54320697-f707bd00-45fe-11e9-802b-0a11dc703057.png)

i'm not asking you for help, i now that the correct template is:
https://yandex.ru/images/search?url={searchTerms}&rpt=imageview
but the "add custom search" feature detects it wrong:
https://yandex.ru/images/?cbir-url={searchTerms}

i post this only because you asked it, maybe it'll be useful to development of the project

do you need such examples further?

**Comment by ssborbis** [comment]
@githubeing 

Hmm, when I try adding a custom engine from https://yandex.ru/images/ I get `https://yandex.ru/images/search?text={searchTerms}` and everything works fine. Maybe a region / locale / language difference in the form url?

**Comment by githubeing** [comment]
you've just added the "find image by text" template, and i was talking about "find image by url"

notice the screenshot above

you have to click the camera button first, then try to add the form "Введите адрес картинки"

**Comment by ssborbis** [comment]
@githubeing K I'll take a look

**Comment by githubeing** [comment]
see screencast with the wrong detection and how i fix it below:
[screencast-2019-03-14_05.32.45.zip](https://github.com/ssborbis/ContextSearch-web-ext/files/2964466/screencast-2019-03-14_05.32.45.zip)

**Comment by ssborbis** [comment]
@githubeing 

A bit off the original issue... I'm running into a bug where I can **only** get the context menu option `Add Custom Search` for the image url search if I **first** bring up the context menu on the main search bar. If I reload the page, click the camera, and try to go straight to the image url search, the `Add Custom Search` menu entry fails to load. It's very bizarre. Are you seeing that too?

**Comment by ssborbis** [comment]
@githubeing 

I posted before I watched your video. Yep, that's what I'm seeing

**Comment by ssborbis** [comment]
@githubeing 

Ok, I've got 3 issues going on.

1. My MutationObserver for dynamically adding listeners to `INPUT` elements wasn't seeing children of added nodes, and missed seeing the image url search bar added to the page when clicking the camera. _**Fixed**_

2. The `Add Custom Search` menu entry was **not** being removed from the context menu properly, meaning it could be added by the first search bar and hang around to be used on the second image url search bar. This was also causing a 2-tiered menu in text boxes when trying to search from the context menu in some cases. _**Fixed**_

3. Yandex using some irregular javascript + form to build the search params. I might not be able to find a generic workaround for this, but getting the first 2 bugs fixed was worth it.

**Comment by githubeing** [comment]
on emojipedia.org the `add custom search` item doesn't show up at all in the context menu almost all the time. but sometimes it does. very rarely, so rarely that it becomes really hard to add the search engine because of this (not using manual adding of course). see screenshots
![Screenshot from 2019-03-21 20-47-39](https://user-images.githubusercontent.com/29702626/54773443-a4d82480-4c1a-11e9-8ce0-b1d15df48d68.png)
![Screenshot from 2019-03-21 20-47-15](https://user-images.githubusercontent.com/29702626/54773444-a4d82480-4c1a-11e9-8c46-ceb9c7bebb4e.png)

**Comment by ssborbis** [comment]
@githubeing 

I'm showing the menu every time, chrome and FF. What's your hardware?

**Comment by ssborbis** [comment]
@githubeing 

Also, do you get better results if you left-click to focus on the input first, then right-click?

**Comment by githubeing** [comment]
not sure what you mean by "what's your hardware". what steps do you expect me to do to answer this your question? my hardware is a desktop pc.

no, i don't get better results when focusing by left-clicking first - i get same results

**Comment by ssborbis** [comment]
I mean what's your processor, ram. I'm trying to figure out if I'm getting different results because of a difference in processing speeds. There could be a timing issue with code execution.

**Comment by ssborbis** [comment]
@githubeing 

I'm seeing it now. This might be related to the `Search with ... ` issue as well. I'll keep you posted

**Comment by githubeing** [comment]
cpu isn't very slow, ram is enough imo, take a look:
![Screenshot from 2019-03-21 21-43-22](https://user-images.githubusercontent.com/29702626/54777015-66defe80-4c22-11e9-82db-2c1e69ff53c8.png)
do you think this could be the cause?

**Comment by ssborbis** [comment]
> cpu isn't very slow, ram is enough imo, take a look:
> do you think this could be the cause?

There's a bit of tricky timing I'm having to deal with to make the correct menus show up in the context menu, but I should be able to find a suitable workaround. I'm rewriting that bit of code and you can try it out when I release the next patch. I'm still not getting the menus to work 100% on my machine, but it's much better than before.

**Comment by githubeing** [comment]
> any sites / forms that don't work with the Add Custom Search function

another site: https://context.reverso.net
the SE would be e.g. [GET] https://context.reverso.net/перевод/русский-английский/{searchTerms}

**Comment by ssborbis** [comment]
@githubeing Thanks. Got it working with a simple fix. I'm sure it was causing failure elsewhere

**Comment by laymanal** [comment]
https://www.bilibili.com/
https://www.blu-ray.com/
https://subhd.com/
https://www.zhihu.com/

**Comment by ssborbis** [comment]
@laymanal 

Several things going on here. The dropdown lists can cause failure if they have any option selected. Clicking an item from the dropdown may skip filling in the form with the search terms and prevents my script from parsing the search terms from the url. Also, some engines replace spaces with +'s which wasn't accounted for. I've added some regex to check for a couple common formats.

> https://www.bilibili.com/

Opens a new tab to search results causing the failure. If you `Add Custom Search` from the results page, you can add the engine

**Comment by Zack11111** [comment]
I can not create Custom Search in bitru.org. Could anyone help me to add this search engine http://bitru.org/browse.php?s=&sort=seeders&tmp=&cat=&subcat=&year=&country=&sound=&soundtrack=&subtitles=#content

**Comment by githubeing** [comment]
> I can not create Custom Search in bitru.org. Could anyone help me to add this search engine http://bitru.org/browse.php?s=&sort=seeders&tmp=&cat=&subcat=&year=&country=&sound=&soundtrack=&subtitles=#content

- right click the search input, 
- click `Add custom search engine`, 
- click `Advanced`
- in the `Template` field, replace the text to: `http://bitru.org/browse.php?s={searchTerms}&sort=seeders#content`
- click `Add`

**Comment by githubeing** [comment]
> I can not create Custom Search in bitru.org. 
> http://bitru.org/browse.php?s=&sort=seeders&tmp=&cat=&subcat=&year=&country=&sound=&soundtrack=&subtitles=#content

@ssborbis , i confirm that the fragment uri part prevents the extension to correctly detect the search uri, and it detects it to be: `http://bitru.org/browse.php#content?s={searchTerms}&sort=seeders` thus making the search prompt part of the fragment which is obviously wrong

**Comment by Zack11111** [comment]
> 
> 
> > Could anyone help me to add this search engine http://bitru.org/browse.php?s=&sort=seeders&tmp=&cat=&subcat=&year=&country=&sound=&soundtrack=&subtitles=#content
> 
>     * right click the search input,
> 
>     * click `Add custom search engine`,
> 
>     * click `Advanced`
> 
>     * in the `Template` field, replace the text to: `http://bitru.org/browse.php?s={searchTerms}&sort=seeders#content`
> 
>     * click `Add`

Thanks a lot. It helped.

**Comment by ssborbis** [comment]
@githubeing 

Yep, I saw the same thing: the hash messes up the template. You posted with the correct template a few seconds before I could. Probably a pretty rare failure but I'll see if there's an easy workaround.

**Comment by stefan5588** [comment]
First, many thanks for your work, the extension is really great! Here is a tough site: http://www.freebookspot.es/. It uses hashes/tokens, so I tried with Search Code, but no success. Any ideas?

**Comment by ssborbis** [comment]
@stefan5588 

What's your template and search code now? Are you searching by title?

**Comment by molkes** [comment]
Website: [Devil Game](https://devilgame.org/)

If the POST method is used, the `_xfToken` of "POST params" will expire after a period of time.

**Comment by githubeing** [comment]
it uses simple request params:
```
TTitleSearch: a title
TAuthorSearch: an author
TISBNSearch: an isbn
DDLLang: English
BSearch: Go
```
also, it has a "text mode":
http://www.freebookspot.es/textmodedefault.aspx?Keyword=an+author

**Comment by ssborbis** [comment]
@molkes 

Yep, gonna need to use the Search Code field if there's an expiring token.

```
let input = document.querySelector('input[name="keywords"]');
input.value = searchTerms;
input.form.submit();
```

**Comment by stefan5588** [comment]
@ssborbis, wow, it works! Thanks!

**Comment by liraqb** [comment]
Website: https://searx.neocities.org/

Trouble you again.

**Comment by ssborbis** [comment]
@liraqb 

Template: https://searx.neocities.org

Search Code
```
let input = document.querySelector('#q');
input.value = searchTerms;
document.querySelector('#go').click();
```

**Comment by maxelo** [comment]
Love this extension, thanks for your work!
this is a chilean news site, that I can't retrieve any url from their search bar

https://www.biobiochile.cl/

(Buscar por = Search for)

![biobio](https://user-images.githubusercontent.com/1003307/85895962-9be2e500-b7c5-11ea-984a-3f5a8344a890.png)

(Buscar por = Search for)

**Comment by ssborbis** [comment]
@maxelo 

You'll need to use the Search Code field.

Name: `biobiochile`
Template: `https://www.biobiochile.cl/`
Icon URI: `https://www.biobiochile.cl/favicon.ico`
Form Path: `https://www.biobiochile.cl`
Search Code:
```
document.querySelector('#search-button2').click();
let input = document.querySelector('input[type="search"]');
input.value = searchTerms;
input.parentNode.querySelector('button').click();
```

**Comment by maxelo** [comment]
thank you very much! it works! =)

Now....because the way the web is designed, I don't think that there's a way to create url search link like this? right?
https://www.biobiochile.cl/search?q=%

....right?

**Comment by ssborbis** [comment]
@maxelo 

Probably not. I haven't dug through that particular site much, but my guess is no.

**Comment by liraqb** [comment]
The difficulty is that you must click the search button to enter keywords.

https://nicemoe11.com/

**Comment by ssborbis** [comment]
@liraqb 

```
(async () => {
document.querySelector('[title="搜索"]').click();
await new Promise(r => setTimeout(r, 1000));
let input = document.querySelector('form > input');
input.value = searchTerms;
input.dispatchEvent(new KeyboardEvent('keypress', {keyCode:13, key:'Enter'}));
})();
```

**Comment by liraqb** [comment]
@ssborbis 
There seems to be a problem. The Enter key is not automatically entered. And pressing Enter will use the default text instead of the entered text.

**Comment by ssborbis** [comment]
@liraqb 

You may need to adjust the delay or even add a delay between `input.value = searchTerms;` and the keyboard event. As written, it works fine for me

**Comment by liraqb** [comment]
@ssborbis 

The selector has not changed. I tried an English browser and turned off other browser plug-ins.

I found that the `value` can only be changed after typing the text, `input.value = searchTerms;` may be invalid, so I try to simulate the input Spacebar to change the `value` , but the analog input is not valid.

The analog input Enter is invalid, but it can be replaced with `document.querySelector('[type="submit"]').click();` .

**Comment by ssborbis** [comment]
@liraqb 
I didn't have any problem simulating the Enter key to submit using KeyboardEvent, but you're right, the form is not accepting a `.value = ` statement to set the search term. There must be an event listener for some other action I'm not seeing.

**Comment by liraqb** [comment]
https://www.uptrends.com/tools/uptime

The search bar is embedded in the current page, but it's actually on another page. You can open the URL in the search bar to query, but the results are not displayed completely.

**Comment by ssborbis** [comment]
@liraqb 

I see, they are using an iframe. It looks like for this particular page, you can set the iframe page URL as your search template / form path and use the searchCode field from there, if you don't mind losing access to the main page.

https://app.uptrends.com/Tools/http?lang=en

**Comment by liraqb** [comment]
@ssborbis 

> https://app.uptrends.com/Tools/http?lang=en

There is no scroll bar on this page, so you can only zoom in to see the full result. I gave up, thank you.

**Comment by ssborbis** [comment]
@liraqb 

Can you make the page scrollable?

`document.body.style.overflow = "auto"`

**Comment by liraqb** [comment]
@ssborbis Oh, this tool is powerful. It worked, thanks.

**Comment by robertheadley** [comment]
https://namecheckup.com/full
no popup "did it work" after search

**Comment by ssborbis** [comment]
@robertheadley 

On the latest dev code, I did get the fail message. It took about 5 seconds to popup. If you're on a release or beta, you might be seeing something different.

In any case, this seems to work as search code

```
document.querySelector('#serachText').value = searchTerms.replace(/\s+/g, "");
document.querySelector('#searchBtn').click();
```

**Comment by Zack11111** [comment]
> > Could anyone help me to add this search engine http://bitru.org/browse.php?s=&sort=seeders&tmp=&cat=&subcat=&year=&country=&sound=&soundtrack=&subtitles=#content
> 
> * right click the search input,
> * click `Add custom search engine`,
> * click `Advanced`
> * in the `Template` field, replace the text to: `http://bitru.org/browse.php?s={searchTerms}&sort=seeders#content`
> * click `Add`

Hi
I have an issue again in the same site (bitru.org). I want to change word "seeders" to "size" in Template field. But, after saving the edit it works fine just until I restart browser. After restart there is "seeders" again.  Same problem in both Chrome and Firefox. Could you help me to fix the issue?

**Comment by robertheadley** [comment]
I don't believe that site supports sorting by size.

**Comment by robertheadley** [comment]
https://bitru.org/browse.php?s={searchTerms}&sort=size&tmp=&cat=&subcat=&year=&country=&sound=&soundtrack=&subtitles=#content

seems to work fine.

**Comment by ssborbis** [comment]
@Zack11111 

This sounds like a bug. Open a new issue and I'll guide you through some troubleshooting

**Comment by robertheadley** [comment]
https://whatsmyname.app/

**Comment by ssborbis** [comment]
@robertheadley 

Javascript engine.

try this search code

```js
let input = document.querySelector('#targetUsername');
input.value = searchTerms;
document.querySelector('BUTTON.btn.btn-success').click();
```

**Comment by robertheadley** [comment]
> 
> 
> @robertheadley
> 
> Javascript engine.
> 
> try this search code
> 
> ```js
> let input = document.querySelector('#targetUsername');
> input.value = searchTerms;
> document.querySelector('BUTTON.btn.btn-success').click();
> ```

Excellent and if I want to specify the category?

**Comment by ssborbis** [comment]
@robertheadley 

You need to get the dropdown selector and set it to the catagory you want, 
```
allnoporn
art
blog
business
coding
dating
finance
gaming
health
hobby
misc
news
political
search
shopping
tech
video
XXXPORNXXX
```
then it appears you need a slight delay before searching

```js
( async () => {
    document.querySelector('#targetUsername').value = searchTerms;
    document.querySelector('#filters').value = 'music';
    await new Promise(r => setTimeout(r, 100));
    document.querySelector('BUTTON.btn.btn-success').click();
})();
```

**Comment by robertheadley** [comment]
A delay of 1000 seems to work thing. Thanks again.

**Comment by yiqixueba56** [comment]
**Template:** https://xmsoushu.com/
Search Code：
`document.querySelector('.el-input__inner').value = searchTerms;
document.querySelector('.el-button.el-button--default').click();`
Using the Search Code above, the website displays "Please enter a keyword" when searching,What should I do in this situation?
![ee36d188a8c7781335471a28b8288d9](https://user-images.githubusercontent.com/102757709/162367243-7a5a92d6-c41d-4435-bde4-37ff842b8b4b.png)

**Comment by ssborbis** [comment]
@yiqixueba56 

I don't know what a valid search term would be, so I can't say for sure if this is working. 

The input is looking for some other events before it registers a proper search. Try the following. 
```js
document.querySelector('.el-input__inner').value = "japan";
document.querySelector('.el-input__inner').dispatchEvent(new Event('input'));
document.querySelector('.el-input__inner').dispatchEvent(new Event('change'));
document.querySelector('.el-button.el-button--default').click();
```

**Comment by yiqixueba56** [comment]
@ssborbis thank you very much! it works!

**Comment by robertheadley** [comment]
> A delay of 1000 seems to work great. Thanks again.

For some reason. This is no longer working.

**Comment by robertheadley** [comment]
> @robertheadley
> 
> You need to get the dropdown selector and set it to the catagory you want,
> 
> ```
> allnoporn
> art
> blog
> business
> coding
> dating
> finance
> gaming
> health
> hobby
> misc
> news
> political
> search
> shopping
> tech
> video
> XXXPORNXXX
> ```
> 
> then it appears you need a slight delay before searching
> 
> ```js
> ( async () => {
>     document.querySelector('#targetUsername').value = searchTerms;
>     document.querySelector('#filters').value = 'music';
>     await new Promise(r => setTimeout(r, 100));
>     document.querySelector('BUTTON.btn.btn-success').click();
> })();
> ```

For some reason, this is no longer working.

**Comment by ssborbis** [comment]
@robertheadley 

I'm not seeing any issues. Tested on FF v101 and Chromium v102

Does it work manually?

**Comment by justsome12** [comment]
can anybody help with this site for mp3 making from youtube links:
https://getn.topsandtees.space

**Comment by ssborbis** [comment]
@justsome12 

This will get you through the first step. You should probably be using something like yt-dlp with an app launcher though for this kind of work.

```javascript
  let form = document.querySelector('form');
  form.q.value = searchTerms;
  form.submit();
```

**Comment by justsome12** [comment]
> yt-dlp

thanks seems interesting, but a bit over the top, I just like to download a song that i like as audio sometimes with a short click on your awesome addon. works. thank you a lot!

**Comment by ssborbis** [comment]
> How can I make it search for the current Url, if the search field is left empty?
/^$/{url}/g
doesn't work here.

Currently, it only works with app launcher commands, but I'll add it to search engines.

**Comment by justsome12** [comment]
> @justsome12
> 
> This will get you through the first step. You should probably be using something like yt-dlp with an app launcher though for this kind of work.
> 
> ```js
>   let form = document.querySelector('form');
>   form.q.value = searchTerms;
>   form.submit();
> ```

offtopic but can I convert this to a bookmarklet for easier access?

**Comment by ssborbis** [comment]
> offtopic but can I convert this to a bookmarklet for easier access?

If you mean a CS script, yes. If you mean a bookmarklet run from Bookmarks, no. At least, not without added code. It wouldn't know what `searchTerms` is. You could use some custom code to get the current selected text or link though.

**Comment by RXUARJ** [comment]
https://www.jiumodiary.com/

**Comment by ssborbis** [comment]
@RXUARJ 

try

```javascript
    document.querySelector('#SearchWord').value = searchTerms;
    document.querySelector('#SearchButton').click();
```

**Comment by RXUARJ** [comment]
> 

Thank you very much

**Comment by Wassss** [comment]
https://free-mp3-download.net

**Comment by ssborbis** [comment]
@Wassss 

try
```javascript
let input = document.querySelector('input');
input.value = searchTerms;
input.dispatchEvent(new Event('change', {bubbles:true}));
document.querySelector('button').click();
```

**Comment by Wassss** [comment]
> @Wassss
> 
> try
> 
> ```js
> let input = document.querySelector('input');
> input.value = searchTerms;
> input.dispatchEvent(new Event('change', {bubbles:true}));
> document.querySelector('button').click();
> ```

@ssborbis Thank you it work !

**Comment by decembre** [comment]
I want mark a serach plugin for:

**Comment by ssborbis** [comment]
@decembre 

Going to https://www.tv5mondeplus.com/fr/search redirects me to https://www.tv5unis.ca/chaines/tv5mondeplus. Is that true for you as well, or am I seeing a regional issue?

**Comment by ssborbis** [comment]
@decembre 

1. you have a typo in line 1 ( missing closing ' )
2. it looks like you also need to dispatch an `input` event

```javascript
  let input = document.querySelector('input[class^="css-1"]');
  input.value = searchTerms;
  input.dispatchEvent(new Event('input', {bubbles:true}));
  input.dispatchEvent(new KeyboardEvent('keydown', {key:"Enter", keyCode:13, bubbles:true}));
  input.dispatchEvent(new KeyboardEvent('keyup', {key:"Enter", keyCode:13, bubbles:true}));
```

**Comment by decembre** [comment]
<blockquote>or am I seeing a regional issue?</blockquote>
yes..... seems you are in Tunisia  ?

The search seems working with your solution...
Thanks ;-)

<blockquote>you also need to dispatch an input event</blockquote>
Where to find infos on these event usable for a search?
How to see them in dev tool ?

**Comment by ssborbis** [comment]
> Where to find infos on these event usable for a search? How to see them in dev tool ? 

Yeah, you can use the dev tools and either inspect the events on elements in the DOM hierarchy, or use the debugger panel to log keyboard events as you perform a normal search.

**Comment by decembre** [comment]
Thanks .
Difficult to find for me.
I google "input.dispatchEvent" /  "KeyboardEvent" / "bubbles" and find some info on them:
I understand now a little the way of the action:
add an event on input and pass an action

But ... i am not a big dev'
:-)

**Comment by Wassss** [comment]
https://snowfl.com/

**Comment by ssborbis** [comment]
@Wassss 

```javascript
document.querySelector('input').value = searchTerms;
document.querySelector('#btn-search').click();
```

**Comment by Wassss** [comment]
> @Wassss
> 
> ```js
> document.querySelector('input').value = searchTerms;
> document.querySelector('#btn-search').click();
> ```

work great thank you !

**Comment by 2ndHouse** [comment]
https://www.dictionary.com/

**Comment by ssborbis** [comment]
@Wassss 

try 
```javascript
 document.querySelector('input').value = searchTerms;
 document.querySelector('input').dispatchEvent(new Event("input"));
 document.querySelector('button').click();
```

My browser skips the CAPTCHA, but you might run into issue there.

**Comment by Wassss** [comment]
> @Wassss
> 
> try
> 
> ```js
>  document.querySelector('input').value = searchTerms;
>  document.querySelector('input').dispatchEvent(new Event("input"));
>  document.querySelector('button').click();
> ```
> 
> My browser skips the CAPTCHA, but you might run into issue there.

Work like a charm, thank you.

**Comment by Leo-Multipotentilite** [comment]
Would you please help me with https://www.jiumodiary.com/
I DID read the previous post and try to add to my search engine by myself for couple hours, but still don't work..
It's really hard for a NOT programmer....

**Comment by ssborbis** [comment]
@Leo-Multipotentilite 

Something like this in the script

```javascript
(async () => {
await new Promise(r => setTimeout(r,1000));
let input = document.querySelector("#SearchWord");
let button = document.querySelector("#SearchButton");
input.value = searchTerms;
button.click();
})();
```

**Comment by Leo-Multipotentilite** [comment]
> [@Leo-Multipotentilite](https://github.com/Leo-Multipotentilite)
> 
> Something like this in the script
> 
> (async () => {
> await new Promise(r => setTimeout(r,1000));
> let input = document.querySelector("#SearchWord");
> let button = document.querySelector("#SearchButton");
> input.value = searchTerms;
> button.click();
> })();
It works. Thank you very much, especialy for the fast response. 
These script helps me a lot, not to mention your add-on.
I couldn't find the donation access on the page, would you accept that? (If you don't mind)
